### PR TITLE
feat(annotations): add support for passed annotations options

### PR DESCRIPTION
**Pull request by mickr**
* Add support for annotation options
* Use messages and locale if preset in options and forward them to the annotator to populate messages and locale

### CODE REVIEW

File: src/lib/viewers/BaseViewer.js [hunk 1]
```diff
@@ -931,17 +931,29 @@ class BaseViewer extends EventEmitter {
...
+        const boxAnnotations =
+            this.options.boxAnnotations || new global.BoxAnnotations(viewerOptions, intl.createAnnotatorIntl());
         this.annotatorConf = boxAnnotations.determineAnnotator(this.options, this.viewerConfig);
 
         if (!this.annotatorConf) {
             return;
         }
 
-        const annotatorOptions = this.createAnnotatorOptions({
+        const options = {
```
Comment (jstoffan) [inline src/lib/viewers/BaseViewer.js:942]:
```
const { intl: intlProvider } = boxAnnotations.getOptions ? boxAnnotations.getOptions() : {};
const annotatorOptions = this.createAnnotatorOptions({
    annotator: this.annotatorConf,
    intl: intlProvider || intl.createAnnotatorIntl(),
    modeButtons: ANNOTATION_BUTTONS
});
```

File: src/lib/viewers/BaseViewer.js [hunk 2]
```diff
@@ -931,17 +931,29 @@ class BaseViewer extends EventEmitter {
...
             annotator: this.annotatorConf,
             modeButtons: ANNOTATION_BUTTONS,
-        });
+        };
+
+        if (this.areNewAnnotationsEnabled() && boxAnnotations.getOptions) {
+            const annotationsOptions = boxAnnotations.getOptions();
+
+            if (annotationsOptions) {
+                const { locale, messages } = annotationsOptions;
```
Comment (jstoffan) [inline src/lib/viewers/BaseViewer.js:951]: This seems a little confusing. Is `language` omitted intentionally? Can we simplify the options logic?

```
const { intl } = boxAnnotations.getOptions ? boxAnnotations.getOptions() : {};
const annotatorOptions = this.createAnnotatorOptions({
    annotator: this.annotatorConf,
    intl: intl || intlUtil.createAnnotatorIntl(),
    modeButtons: ANNOTATION_BUTTONS
});
```

File: src/lib/viewers/__tests__/BaseViewer-test.js [hunk 3]
```diff
@@ -1448,8 +1499,6 @@ describe('lib/viewers/BaseViewer', () => {
             expect(combinedOptions.location).to.deep.equal({ locale: 'en-US' });
             expect(combinedOptions.randomOption).to.equal('derp');
             expect(combinedOptions.localizedStrings).to.not.be.undefined;
-            expect(combinedOptions.intl.language).to.equal('en-US');
```
Comment (jstoffan) [inline src/lib/viewers/__tests__/BaseViewer-test.js:1451]: I think we'll want more tests for the final output of the options object.

File: src/lib/viewers/BaseViewer.js (hunk 2 above)
Comment (mickr) [reply src/lib/viewers/BaseViewer.js:951]: Sure, I passed it to the boxAnnotations instantiation on purpose to be explicit how it should be used.  This should be fine as well. Thanks for the input

File: src/lib/viewers/__tests__/BaseViewer-test.js (hunk 3 above)
Comment (mickr) [reply src/lib/viewers/__tests__/BaseViewer-test.js:1451]: I do check in the `createAnnotator()` test  that this gets called with the correct properties. I have added back the check for language as well.

File: src/lib/i18n.js [hunk 4]
```diff
@@ -7,6 +7,10 @@ try {
 
 const language = __LANGUAGE__ || 'en-US'; // eslint-disable-line
 
+const getLocale = lang => {
+    return lang.substr(0, lang.indexOf('-'));
```
Comment (ConradJChan) [inline src/lib/i18n.js:11]: Do we need to handle the case where the lang doesn't have a dash?

Comment (mickr) [reply src/lib/i18n.js:11]: I believe all of our supported langs have a dash.

**Pull request merged**